README.md

```diff
@@ -1,44 +1,235 @@
+
+**Select Mode**
+
+Select mode describes which select mode the data source is in.
+
+**getState**
+
+`getState` is used to retrieve the current state of the dashboard. The state format is documented [here](https://help.salesforce.com/articleView?id=sf.bi_embed_filters.htm&type=5).
+
+**setState**
+
+`setState` is used to patch the current state of the dashboard.
+
+## Tableau CRM LWC Metadata
+
+Lightning Web Components used in Tableau CRM dashboards leverage the same file structure and [XML metadata format](https://developer.salesforce.com/docs/component-library/documentation/en/lwc/lwc.reference_configuration_tags) used by other LWCs on the Salesforce platform. Here are few additions to this metadata specific for Tableau CRM.
+
+**analytics__Dashboard Target**
+
+Adding this target to the list of `<targets>` for your component allows it to be used in Tableau CRM dashboards, provided that your component is public. You can also add a `<targetConfig>` for the new target to further customize how your component appears in dashboards.
+
+**`<hasStep>` Tag**
+
+In an `analyticsDashboard` target config, you can choose to include a `<hasStep>true</hasStep>` tag to indicate that your component requires an attached step to function as expected. With this tag, the dashboard builder UI prompts you to attach an existing step or create a new step when creating an instance of your component. Components with an attached step have access to step-specific properties like `results` and `selection`. For more information on these properties, see [Tableau CRM LWC Features](#tableau-crm-lwc-features).
+
+**Measure and Dimension Attribute Data Types**
+
+Attributes specified in a LWC's `analtyicsDashboard` target config are displayed in the Tableau CRM dashboard builder UI for configuration. In addition to the common data types, this target also supports `Measure` and `Dimension` data types for components with `<hasStep>true</hasStep>`. Dashboard builders are able to choose a column of the given data type from the results of the attached step.
+
+## Tableau CRM App Templates
+
+Use the Quick Start template to practice working with Tableau CRM app templates in your scratch org. Then, create
+your own apps and templatize them using the Analytics CLI plugin and Visual Studio commands.
+
+Navigate to the sfdx-analytics directory in a terminal:
+
+1. Update the adminEmail in config/project-scratch-def.json to your email address
+
+    ```
+    sfdx force:org:open -u mydevorg
+    ```
+
+1.  Authenticate to your dev hub.
+    ```
+    sfdx force:auth:web:login -d
+    ```
+
+1.  Create a scratch org that is Analytics enabled.
+    ```
+    sfdx force:org:create -s -f config/project-scratch-def.json
+    ```
+
+1.  Push the sample Analytics template from the local workspace to the scratch org
+    ```
+    sfdx force:source:push
+    ```
+
+1. Install the Analytics SFDX plugin
+    ```
+    sfdx plugins:install @salesforce/analytics
```

Comment:
If this whole section is combined with the above section, this should be added there, too.